frontend/test/__support__/server-mocks/database.ts

```diff
@@ -1,16 +1,31 @@
 import { Scope } from "nock";
 import _ from "underscore";
 import { Database } from "metabase-types/api";
+import { SAVED_QUESTIONS_VIRTUAL_DB_ID } from "metabase-lib/metadata/utils/saved-questions";
 import { setupTableEndpoints } from "./table";
 
+const SAVED_QUESTIONS_DB = {
+  id: SAVED_QUESTIONS_VIRTUAL_DB_ID,
+  name: "Saved Questions",
+  is_saved_questions: true,
+};
+
 export function setupDatabaseEndpoints(scope: Scope, db: Database) {
   scope.get(`/api/database/${db.id}`).reply(200, db);
   setupSchemaEndpoints(scope, db);
   db.tables?.forEach(table => setupTableEndpoints(scope, table));
 }
 
-export function setupDatabasesEndpoints(scope: Scope, dbs: Database[]) {
+export function setupDatabasesEndpoints(
+  scope: Scope,
+  dbs: Database[],
+  { hasSavedQuestions = true } = {},
```

Comment:
How does it work without declaring a type? It seems that this directory is not type-checked.

Reply:
Also, should we make it false? I'd like the test to be explicit that it relies on saved questions to be present

Reply:
> How does it work without declaring a type? It seems that this directory is not type-checked.

@ranquild, I think the problem is that we expect `dbs` arg to be `Database[]`, but there's no issue about merging them with the saved questions database object inside `nock().reply`

> Also, should we make it false? I'd like the test to be explicit that it relies on saved questions to be present

Yeah, on the other hand, it's very easy to forget we need to handle saved questions database here and there. I don't have a strong opinion though, and happy tweak it.

While we're here, what do you think about exporting the `SAVED_QUESTIONS_DATABASE` object and adding it to the database list manually on a need? Example:

```tsx
import { setupDatabasesEndpoints, SAVED_QUESTIONS_DATABASE } from "__support__/server-mocks"

setupDatabasesEndpoints(scope, [
  createMockDatabase(),
  SAVED_QUESTIONS_DATABASE,
]);

Reply:
I don't think we should export constants from server-mocks. It was never intended to start defining mock data there. WDYT about adding a constant in the app itself, similar to `ROOT_COLLECTION`? https://github.com/metabase/metabase/blob/62498154be25ca45e014731ff39894c43cfa874c/frontend/src/metabase/entities/collections/constants.ts#L5

One thing is that you would need to call `createMockDatabase(SAVED_QUESTIONS_DATABSE)`

Reply:
@ranquild moved `SAVED_QUESTIONS_DATABASE` to `metabase/databases/constants`, similar to how we deal with `ROOT_COLLECTION`.

Also, after looking into this a little bit more, I think I have two strong opinions:

1. We shouldn't wrap `SAVED_QUESTIONS_DATABASE` into `createMockDatabase`. When requesting a saved questions database from the BE, it comes with `id`, `name`, `is_saved_questions`, and `features` parameters. `createMockDatabase` will add a bunch of other stuff, but essentially we'd end up having an object we're not actually dealing with in a live app
2. `hasSavedQuestions` should be `true` by default. The BE only returns the saved question database when we call `/api/database` with the `?saved=true` parameter, i.e. regular `GET /api/database` response doesn't have it. IMO, if we're testing a piece of code explicitly asking for databases _including_ the saved questions one, then it should be included in mock data by default. Passing `hasSavedQuestions: false` should rather be used to test how we handle "no saved question on the instance scenario" IMO

Does this make sense?

Reply:
re 1. I think it breaks our type system because it makes `SAVED_QUESTIONS_DATABASE` not be compatible with `Database` from `metabase-types/api`. I'm ok with not using `createMockDatabase`, but in that case please add `Database` type to the constant and add required fields

Reply:
If some fields aren't present for this db in the backend response, we should modify `Database` type